### PR TITLE
add SA role bindings to auto-provisioned namespaces

### PR DESCRIPTION
**Pull request by deads2k**
The `openshift` and `openshift-infra` namespaces were missing the rolebindings for their default service accounts.  That lack of consistency has led to questions about creating builds in the `openshift` namespaces.

I don't feel strongly about `openshift-infra` since that one is "ours", but I like the consistency of having both.

@liggitt

### CODE REVIEW

File: pkg/cmd/server/origin/ensure.go [hunk 1]
```diff
@@ -93,57 +98,57 @@ func (c *MasterConfig) ensureComponentAuthorizationRules() {
 	}
 }
 
-// ensureDefaultNamespaceServiceAccountRoles initializes roles for service accounts in the default namespace
-func (c *MasterConfig) ensureDefaultNamespaceServiceAccountRoles() {
+// ensureNamespaceServiceAccountRoleBindings initializes roles for service accounts in the namespace
+func (c *MasterConfig) ensureNamespaceServiceAccountRoleBindings(namespaceName string) {
 	const ServiceAccountRolesInitializedAnnotation = "openshift.io/sa.initialized-roles"
 
 	// Wait for the default namespace
```
Comment (liggitt) [inline pkg/cmd/server/origin/ensure.go]: loop probably doesn't make sense for anything other than default

Comment (deads2k) [inline pkg/cmd/server/origin/ensure.go]: > loop probably doesn't make sense for anything other than default

You'd rather it factored separately?  I don't feel strongly.

File: pkg/cmd/server/origin/ensure.go [hunk 2]
```diff
@@ -32,7 +32,10 @@ func (c *MasterConfig) ensureOpenShiftSharedResourcesNamespace() {
 		_, err = c.KubeClient().Namespaces().Create(namespace)
 		if err != nil {
 			glog.Errorf("Error creating namespace: %v due to %v\n", namespace, err)
+			return
 		}
+
+		c.ensureNamespaceServiceAccountRoleBindings(c.Options.PolicyConfig.OpenShiftSharedResourcesNamespace)
```
Comment (liggitt) [inline pkg/cmd/server/origin/ensure.go]: three references... save `c.Options.PolicyConfig.OpenShiftSharedResourcesNamespace` to a local var?

File: pkg/cmd/server/origin/ensure.go (hunk 1 above)
Comment (liggitt) [inline pkg/cmd/server/origin/ensure.go]: I wonder if we should wait for the default namespace before calling this function

Comment (deads2k) [inline pkg/cmd/server/origin/ensure.go]: > I wonder if we should wait for the default namespace before calling this function

That would be the refactor.  Sure.

**Comment by deads2k** [comment]
comments addressed.

**Comment by liggitt** [comment]
lgtm [merge]

**Pull request merged**